app/src/main/java/org/wikipedia/language/FirebaseLanguageDetector.kt

```diff
@@ -0,0 +1,20 @@
+package org.wikipedia.language
+
+import com.google.mlkit.nl.languageid.LanguageIdentification
+
+class FirebaseLanguageDetector {
+    interface Callback {
+        fun onLanguageDetectionSuccess(languageCode: String)
+    }
+
+    var callback: Callback? = null
+    fun detectLanguageFromText(text: String) {
+        val languageIdentifier = LanguageIdentification.getClient()
+        languageIdentifier.identifyLanguage(text)
+                .addOnSuccessListener { languageCode: String ->
+                    if (callback != null) {
```

Comment:
A null check is not necessary. You can just use `callback?.onLanguageDetectionSuccess()`